data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetAttestationData.java

```diff
@@ -89,31 +113,27 @@ public GetAttestationData(final ValidatorDataProvider provider, final JsonProvid
         @OpenApiResponse(status = RES_SERVICE_UNAVAILABLE, description = SERVICE_UNAVAILABLE)
       })
   @Override
-  public void handle(Context ctx) throws Exception {
-    try {
-      final Map<String, List<String>> parameters = ctx.queryParamMap();
-      if (parameters.size() < 2) {
-        throw new IllegalArgumentException(
-            String.format("Please specify both %s and %s", SLOT, COMMITTEE_INDEX));
-      }
-      UInt64 slot = getParameterValueAsUInt64(parameters, SLOT);
-      int committeeIndex = getParameterValueAsInt(parameters, COMMITTEE_INDEX);
-      if (committeeIndex < 0) {
-        throw new IllegalArgumentException(
-            String.format("'%s' needs to be greater than or equal to 0.", COMMITTEE_INDEX));
-      }
+  public void handle(@NotNull Context ctx) throws Exception {
+    adapt(ctx);
+  }
 
-      final SafeFuture<Optional<AttestationData>> future =
-          provider.createAttestationDataAtSlot(slot, committeeIndex);
-      handleOptionalResult(ctx, future, this::processResult, SC_BAD_REQUEST);
-    } catch (final IllegalArgumentException e) {
-      ctx.json(jsonProvider.objectToJSON(new BadRequest(e.getMessage())));
-      ctx.status(SC_BAD_REQUEST);
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final UInt64 slot = request.getQueryParameter(SLOT_PARAM);
+    final int committeeIndex = request.getQueryParameter(COMMITTEE_INDEX_PARAMETER_INT);
+    if (committeeIndex < 0) {
+      throw new IllegalArgumentException(
+          String.format("'%s' needs to be greater than or equal to 0.", COMMITTEE_INDEX));
     }
```

Comment:
we might actually be better off explicitly just `request.respondError` here, rather than throwing and making the default handler figure it out...